src/main/java/org/opensearch/knn/index/codec/jvector/JVectorReader.java

```diff
@@ -92,19 +87,15 @@ public JVectorReader(SegmentReadState state, boolean mergeOnDisk) throws IOExcep
 
     @Override
     public void checkIntegrity() throws IOException {
-        flatVectorsReader.checkIntegrity();
         for (FieldEntry fieldEntry : fieldEntryMap.values()) {
-            try (var indexInput = state.directory.openInput(fieldEntry.vectorIndexFieldDataFileName, state.context)) {
+            try (var indexInput = state.directory.openInput(fieldEntry.vectorIndexFieldDataFileName, IOContext.READONCE)) {
                 CodecUtil.checksumEntireFile(indexInput);
             }
         }
     }
 
     @Override
     public FloatVectorValues getFloatVectorValues(String field) throws IOException {
-        if (mergeOnDisk) {
```

Comment:
Can we split this for a separate change so it's easier to validate any issues?
I think maybe let's make current change add the PQ parameter as originally stated in the description, for a separate change we can evaluate the impact of going back to the pure inline jVector values.